policies/test.rego

```diff
@@ -21,7 +23,7 @@ deny[{"msg": msg}] {
 
 # Check if all tests succeeded 
 deny[{"msg": msg}] {
-	all_failed := [failure | data.test[_].result != "SUCCESS"; failure := 1]
+	all_failed := [failure | [data.test[name].result != "SUCCESS", sprintf("test:%s", [name]) in data.config.policy.non_blocking_checks] == [true, false]; failure := name]
```

Comment:
I've found no way of using `cond1 and !cond2` in a comprehension's condition, so I created `[cond1, cond2]` array and compare it to `[true, false]`

Reply:
This is pretty hard to read for a noob like me. Would it be possible to break the process down a bit? Maybe first get a list of all non-successful tests, then in another statement, filter out non blocking checks?

Reply:
Split into separate steps and added comments.